File: vidyaras_app/lib/src/features/home/3_domain/models/course.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'course.freezed.dart';
part 'course.g.dart';

/// Course domain model
/// Represents a course in the VidyaRas application
/// Aligned with the courses table in Supabase
@freezed
class Course with _$Course {
  const factory Course({
    required String id,
    required String title,
    required String instructor,
    String? thumbnailUrl,
    String? description,
    @Default(0.0) double rating,
    @Default(0) int reviewCount,
    @Default(0) int enrolledCount,
    String? price,
    double? originalPrice,
    String? duration,
    @Default(false) bool isFree,
    @Default(false) bool isLive,
    @Default(false) bool hasFreeTrial,
    @Default(false) bool isRecorded,
    @Default(false) bool hasCertificate,
    @Default(false) bool hasQuizzes,
    @Default(false) bool hasMaterials,
    String? category,
    @Default('Bilingual (Hindi + English)') String language,
    @Default('published') String status, // draft, published, archived
    int? accessDays,
    double? progress, // 0.0 to 1.0 for enrolled courses
    String? createdBy,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) = _Course;

  factory Course.fromJson(Map<String, dynamic> json) => _$CourseFromJson(json);
}

File: vidyaras_admin/src/types/course.ts
export interface Course {
  id: string;
  title: string;
  instructor: string;
  thumbnailUrl?: string;
  rating: number;
  reviewCount: number;
  enrolledCount: number;
  price?: string;
  duration?: string;
  isFree: boolean;
  isLive: boolean;
  hasFreeTrial: boolean;
  isRecorded: boolean;
  progress?: number;
  categories?: string[];
  createdAt?: string;
}

export interface PricingOption {
  fullPrice: number;
  emiMonthlyPrice?: number;
  emiTenure?: number;
  isFree: boolean;
}

export interface BatchInfo {
  id: string;
  name: string;
  startDate: string;
  schedule: string;
  time: string;
  capacity: number;
  enrolled: number;
}

export interface LessonItem {
  id: string;
  title: string;
  durationMinutes: number;
  isLocked: boolean;
  isCompleted: boolean;
  description?: string;
}

export interface SectionInfo {
  id: string;
  title: string;
  lessons: LessonItem[];
  totalDurationMinutes?: number;
}

export interface CourseReview {
  id: string;
  userName: string;
  userAvatar?: string;
  rating: number;
  reviewText: string;
  createdAt: string;
}

export interface CourseDetail {
  basicInfo: Course;
  description: string;
  whatYouLearn: string[];
  courseIncludes: string[];
  prerequisites: string[];
  curriculum: SectionInfo[];
  pricing: PricingOption;
  reviews: CourseReview[];
  liveBatch?: BatchInfo;
  accessDays?: number;
  accessExpiryDate?: string;
  language?: string;
  hasCertificate: boolean;
  hasQuizzes: boolean;
  hasMaterials: boolean;
}

export type CourseType = 'live' | 'recorded' | 'free';

export interface MissingFields {
  critical: string[];
  optional: string[];
}
